sdk/monitor/azure-monitor-opentelemetry-exporter/azure/monitor/opentelemetry/exporter/_storage.py

```diff
@@ -197,7 +237,24 @@ def put(self, data, lease_period=None):
         )
         if lease_period is None:
             lease_period = self._lease_period
-        return blob.put(data, lease_period=lease_period)
+
+        # blob.put should an exception_message if an expection occurs else should return None
+        if self._customer_statsbeat_metrics:
+            result, exception_message = blob.put(data, lease_period=lease_period, return_exception=True)
```

Comment:
I think this might throw an exception if `blob.put` returns normally without an exception (since it returns self which is a `LocalFileBlob`). Any reason why you can't put the statsbeat record logic in `LocalFileBlob.put`?